PopPUNK/__main__.py

```diff
@@ -687,57 +747,71 @@ def assign_query(dbFuncs, ref_db, q_files, output, update_db, full_db, distances
                                                     threads = threads)
         qcPass = qcDistMat(distMat, refList, queryList, max_a_dist)
 
-        # Assign these distances as within or between
-        model_prefix = ref_db
-        if model_dir is not None:
-            model_prefix = model_dir
-        model = loadClusterFit(model_prefix + "/" + os.path.basename(model_prefix) + '_fit.pkl',
-                                model_prefix + "/" + os.path.basename(model_prefix) + '_fit.npz')
-        queryAssignments = model.assign(distMat)
-
-        # set model prefix
-        model_prefix = ref_db
-        if model_dir is not None:
-            model_prefix = model_dir
-
-        # Set directories of previous fit
-        if previous_clustering is not None:
-            prev_clustering = previous_clustering
-        else:
-            prev_clustering = model_prefix
-
-        # Load the network based on supplied options
-        genomeNetwork, old_cluster_file = fetchNetwork(prev_clustering, model, refList,
-                                                        core_only, accessory_only)
-
-        # Assign clustering by adding to network
-        ordered_queryList, query_distMat = addQueryToNetwork(dbFuncs, refList, q_files,
-                genomeNetwork, kmers, estimated_length, queryAssignments, model, output, update_db,
-                use_mash, threads)
-
-        # if running simple query
-        print_full_clustering = False
-        if update_db:
-            print_full_clustering = True
-        isolateClustering = {'combined': printClusters(genomeNetwork, output + "/" + os.path.basename(output),
-                                                        old_cluster_file, external_clustering, print_full_clustering)}
-
-        # update_db like no full_db
-        if update_db:
+        # Calculate query-query distances
+        ordered_queryList = []
+        
+        # Assign to strains or lineages, as requested
+        if assign_lineage:
+
+            # Assign lineages by calculating query-query information
+            ordered_queryList, query_distMat = calculateQueryDistances(dbFuncs, refList, q_files,
+                    kmers, estimated_length, output, use_mash, threads)
+
+        if not assign_lineage:
```

Comment:
else

Reply:
Fixed in 8e0875ff87660230a3b41e195d93e3c2f49464a9.